src/services/services.ts

```diff
@@ -1011,6 +1014,17 @@ module ts {
         isWriteAccess: boolean;
     }
 
+    export interface DocumentHighlights {
+        fileName: string;
+        highlightSpans: HighlightSpan[];
+    }
+
+    export interface HighlightSpan {
+        textSpan: TextSpan;
+        isDefinition: boolean;
+        isWriteAccess: boolean;

```

Comment:
yeah - Monaco differs between read and write access when rendering highlight. 
![image](https://cloud.githubusercontent.com/assets/1794099/7070272/30d55dda-dede-11e4-9cbf-c5e45ce9f67e.png)